src/functions/anti-scam/checkScam.ts

```diff
@@ -8,26 +9,25 @@ import { resolveRedirect } from '../../util/resolveRedirect';
 export async function checkScam(content: string): Promise<string[]> {
 	const redis = container.resolve<Redis>(kRedis);
 
-	const urlRegex =
-		/(http(s)?:\/\/.)?(www\.)?[-a-zA-Z0-9@:%._\+~#=]{2,256}\.[a-z]{2,6}\b([-a-zA-Z0-9@:%_\+.~#?&//=]*)/gi;
+	const linkRegex = /(?:https?:\/\/)(?:www\.)?[-a-zA-Z0-9@:%._\+~#=]{2,256}\.[a-z]{2,6}\b[-a-zA-Z0-9@:%_\+.~#?&//=]*/gi;
 	const scamDomains = await redis.smembers('scamdomains');
 	const trippedDomains = [];
 
 	let matches: any[] | null = [];
-	while ((matches = urlRegex.exec(content)) !== null) {
-		const url = matches[0];
-		const urlLower = url.toLowerCase();
-		const hit = scamDomains.find((d) => urlLower.includes(d.toLowerCase()));
+	while ((matches = linkRegex.exec(content)) !== null) {
+		const url = new URL(matches[0]);
+		const hit = scamDomains.find((d) => url.host.endsWith(d));
 
 		if (hit) {
 			trippedDomains.push(hit);
 			continue;
 		}
 
 		try {
-			const resolved = await resolveRedirect(url);
-			const resolvedLower = resolved.toLowerCase();
-			const hit = scamDomains.find((domain) => resolvedLower.includes(domain));
+			const r = await resolveRedirect(matches[0]);
```

Comment:
You can use `url.href` to get the sanitised URL:
```suggestion
			const r = await resolveRedirect(url.href);
```

Or you can also edit the function to accept an `URL` instance, since most libraries (and even Node.js's HTTP) support it.